core/services/assets/Registry.php

```diff
@@ -70,17 +74,58 @@ class Registry
     private $manifest_data = array();
 
 
+    /**
+     * Holds any dependency data obtained from registered dependency map json.
+     * Dependency map json is generated via the @wordpress/dependency-extraction-webpack-plugin via the webpack config.
+     * @see https://github.com/WordPress/gutenberg/tree/master/packages/dependency-extraction-webpack-plugin
+     *
+     * @var array
+     */
+    private $dependencies_data = [];
+
+    /**
+     * @var DomainInterface
+     */
+    private $domain;
+
+    /**
+     * This is a known array of possible wp css handles that correspond to what may be exposed as dependencies in our
+     * build process.  Currently the dependency export process in webpack does not consider css imports, so we derive
+     * them via the js dependencies (WP uses the same handle for both js and css). This is a list of known handles that
+     * are used for both js and css.
+     * @var array
+     */
+    private $wp_css_handle_dependencies = [
+        'wp-components',
+        'wp-block-editor',
+        'wp-block-library',
+        'wp-edit-post',
+        'wp-edit-widgets',
+        'wp-editor',
+        'wp-format-library',
+        'wp-list-reusable-blocks',
+        'wp-nux',
+    ];
+
+
     /**
      * Registry constructor.
      * Hooking into WP actions for script registry.
      *
-     * @param AssetCollection $assets
-     * @param I18nRegistry    $i18n_registry
+     * @param AssetCollection      $assets
+     * @param I18nRegistry         $i18n_registry
+     * @param DomainInterface|null $domain
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
      */
-    public function __construct(AssetCollection $assets, I18nRegistry $i18n_registry)
+    public function __construct(AssetCollection $assets, I18nRegistry $i18n_registry, DomainInterface $domain = null)
```

Comment:
hmmmm...

the `Registry` is used by ALL `AssetManagers` so have you investigated/considered how this would affect addons that are registering their assets under their own domain?

I looked through things and don't see any potential issues, but just want to be 150% sure this won't cause problems.

It looks like we are only using the domain for `distributionAssetsPath()` which likely will not change for addons, so I think this is fine for now as long as addons won't require their own bundled dependencies (which doesn't make sense).

thoughts?

Reply:
Great catch.  We discussed this in slack and arrived at the conclusion this injection is not needed because `ManifestFile` has the path exposed. See 15b8d47.